smart-connector/src/main/java/eu/knowledge/engine/smartconnector/impl/ReasonerProcessor.java

```diff
@@ -70,6 +77,8 @@ public class ReasonerProcessor extends SingleInteractionProcessor {
 	private final Set<PostExchangeInfo> postExchangeInfos;
 	private Set<Rule> additionalDomainKnowledge;
 	private ReasonerPlan reasonerPlan;
+	//private Set<Set<TriplePattern>> knowledgeGaps;
+	private KnowledgeGapSet knowledgeGaps;
```

Comment:
Two thoughts about the `KnowledgeGapSet` and `KnowledgeGap` classes:
- I would not have created the separate `KnowledgeGapSet` class, because that is not that much of a difference from `Set<KnowledgeGap>` and it is also not very common in Java to do it like that.
- I am in doubt about whether to create the `KnowledgeGap` class, because on the one hand they are `just` graph patterns and we represent those in the reasoner as `Set<TriplePattern>`. Other the other hand, they are not entirely graph patterns because they are typically more spread out over the reasoning graph.

Suggestion: remove the `KnowledgeGapSet` class and keep the `KnowledgeGap` class.

Reply:
@bnouwt : Ok, I created the `KnowledgeGapSet` because we are also using the `BindingSet` in the Smart Connector API. But, I agree that the added value of a separate `KnowledgeGapSet` class is not very high, except for the readability/clarity of the code.

Concluding: I will remove the `KnowledgeGapSet` class and replace with `Set<KnowledgeGap>`